src/sparseml/keras/models/external/keras_applications.py

```diff
@@ -0,0 +1,127 @@
+        )
+        arch, sub_arch = _get_architecture(model_func_name)
+
+        # wrap model constructor for registry compatibility
+        wrapped_constructor = _registry_constructor_wrapper(key, model_func)
+
+        ModelRegistry.register_wrapped_model_constructor(
+            wrapped_constructor,
+            key=key,
+            input_shape=input_shape,
+            domain="cv",
+            sub_domain="classification",
+            architecture=arch,
+            sub_architecture=sub_arch,
+            default_dataset="imagenet",
+            default_desc="base",
+            repo_source="keras.applications",
+        )
+
+
+def _registry_constructor_wrapper(key, model_func):
+    # wraps the keras_applications model constructor function to be compatible
+    # with sparseml model registry loading
+    def wrapper(
+        pretrained_path: str = None,
+        pretrained: Union[bool, str] = False,
+        pretrained_dataset: str = None,
+        **kwargs,
+    ):
+        """
+        :param pretrained_path: A path to the pretrained weights to load,
+            if provided will override the pretrained param
+        :param pretrained: True to load the default pretrained weights,
+            a string to load a specific pretrained weight
+            (ex: base, pruned-moderate),
+            or False to not load any pretrained weights
+        :param pretrained_dataset: The dataset to load pretrained weights for
+            (ex: imagenet, mnist, etc).
+            If not supplied will default to the one preconfigured for the model.
+        """
+        if isinstance(pretrained, str):
+            if pretrained.lower() == "true":
+                pretrained = True
+            elif pretrained.lower() in ["false", "none"]:
+                pretrained = False
+
+        weights = None
+        if pretrained:
+            if pretrained_dataset == "imagenet":
+                weights = "imagenet"  # Imagenet pretrained weights
+            elif pretrained_path is not None:
+                weights = pretrained_path  # Path to a weight file
+        if weights is not None:
+            LOGGER.info("Model being created with weights from {}".format(weights))
+        else:
+            LOGGER.info("Model being created with random weights")
+        try:
+            model = model_func(weights=weights, **kwargs)
+        except ValueError:
+            # Load the model directly assuming "weights" contain all information
+            # about the model
+            LOGGER.info("Loading model directly from {}".format(weights))
+            model = keras.models.load_model(weights)
+        return model
+
+    return wrapper
+
+
+def _get_architecture(model_name):
+    if model_name == "ResNet50":
+        return "ResNet_v1", "50"
+    else:
+        raise ValueError("Model {} unknown or not supported".format(model_name))
+
+
+def main():
+    _register_classification_models()
```

Comment:
we should have a safety here to make sure this can only be called once

Reply:
In register_wrapped_model_constructor of the model registry, we already have a check to throw if an arch key is added the second time.